src/gluonts/model/forecast.py

```diff
@@ -508,42 +511,33 @@ def __repr__(self):
         )
 
 
-class OutputType:
-    values = {'mean', 'samples', 'quantiles'}
-
-    def __init__(self, v):
-        assert (
-            v in OutputType.values
-        ), f'unknown output type {v}, expected one of: {", ".join(OutputType.values)}'
-
-        self.v = v
+class OutputType(str, Enum):
+    mean = 'mean'
+    samples = 'samples'
+    quantiles = 'quantiles'
 
-    def __repr__(self) -> str:
-        return self.v
 
-    def __eq__(self, that):
-        return str(that) == str(self)
+class Config(pydantic.BaseModel):
+    num_eval_samples: int
+    output_types: Set[OutputType]
+    quantiles: List[str]  # FIXME: validate list elements
 
-    def __hash__(self):
-        return hash(str(self))
+    def process(self, forecast: Forecast):
+        result = {}
 
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
+        if OutputType.mean in self.output_types:
+            result['mean'] = forecast.mean.tolist()
 
-    @classmethod
-    def validate(cls, v):
-        if not isinstance(v, str):
-            raise ValueError(
-                f'strict string: expected string value, got {type(v)}'
+        if OutputType.samples in self.output_types:
+            result['samples'] = (
+                forecast.samples.tolist()
+                if hasattr(forecast, 'samples')
```

Comment:
I thought the `get_samples` method introduced above was meant to avoid this `if`?

Reply:
Right. So, do you prefer having a samples method on every Forecast object, or do the logic elsewhere?

Reply:
I don't have strong preferences, I just wanted to make sure this was intended and the `get_samples` is not some left-over code. The code can go as-is